core/verify_manager.go

```diff
@@ -0,0 +1,150 @@
+package core
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/log"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+const verifiedCacheSize = 256
+
+type VerifyManager struct {
+	bc                   *BlockChain
+	tasks                map[common.Hash]*VerifyTask
+	peers                VerifyPeers
+	verifiedCache        *lru.Cache
+	allowUntrustedVerify bool
+	verifyCh             chan common.Hash
+	exitCh               chan struct{}
+}
+
+func NewVerifyManager(blockchain *BlockChain) *VerifyManager {
+	verifiedCache, _ := lru.New(verifiedCacheSize)
+	vm := &VerifyManager{
+		bc:            blockchain,
+		tasks:         make(map[common.Hash]*VerifyTask),
+		verifiedCache: verifiedCache,
+		verifyCh:      make(chan common.Hash),
+		exitCh:        make(chan struct{}),
+	}
+	return vm
+}
+
+func (vm *VerifyManager) Start() {
+	//read disk store to initial verified cache
+	//load unverified blocks in a normalized chain and start a batch of verify task
+	header := vm.bc.CurrentHeader()
+	go vm.mainLoop(header)
+}
+
+func (vm *VerifyManager) Stop() {
+	//stop all the tasks
+	close(vm.exitCh)
+}
+
+func (vm *VerifyManager) mainLoop(header *types.Header) {
```

Comment:
No need to pass `header *types.Header`, since vm hold `blockchain`, we can get currentBlockHeader by `bc`